packages/melos/lib/src/commands/run.dart

```diff
@@ -0,0 +1,211 @@
+      if (choices.length == 1) {
+        // Only 1 package - no need to prompt the user for a selection.
+        selectedPackage = packages[0].name;
+      } else if (noSelect) {
+        // Skipping selection if flag present.
+        selectedPackage = choices[0];
+      } else {
+        // Prompt user to select a package.
+        selectedPackage = prompts.choose(
+          [
+            AnsiStyles.white('Select a package to run the '),
+            AnsiStyles.cyan(script.name),
+            AnsiStyles.white(' script'),
+            AnsiStyles.gray(''),
+          ].join(),
+          choices,
+          interactive: false,
+          defaultsTo: choices[0],
+        )!;
+      }
+
+      final selectedPackageIndex =
+          choices.length > 1 ? choices.indexOf(selectedPackage) : 1;
+      // Comma delimited string of packages selected (all or a single package).
+      final packagesEnv = selectedPackageIndex == 0 && choices.length > 1
+          ? packages.map((e) => e.name).toList().join(',')
+          : packages[selectedPackageIndex - 1].name;
+      // MELOS_PACKAGES environment is detected by melos itself when through
+      // a defined script, this comma delimited list of package names is used
+      // instead of any filters if detected.
+      environment[envKeyMelosPackages] = packagesEnv;
+      logger.stdout('\n');
+    }
+
+    final scriptSource = script.run;
+    final scriptParts = scriptSource.split(' ');
+
+    logger.stdout(AnsiStyles.yellow.bold('melos run ${script.name}'));
+    logger.stdout(
+      '   └> ${AnsiStyles.cyan.bold(scriptSource.replaceAll('\n', ''))}',
+    );
+    logger.stdout('       └> ${AnsiStyles.yellow.bold('RUNNING')}\n');
+
+    return startProcess(
+      scriptParts,
+      environment: environment,
+      workingDirectory: config.path,
+    );
+  }
+}
+
+class NoPackageFoundScriptException implements MelosException {
+  NoPackageFoundScriptException._(this.filter, this.scriptName);
+
+  final PackageFilter? filter;
+  final String? scriptName;
+
+  @override
+  String toString() {
+    return 'NoPackageFoundScriptException: No package found that matches with '
+        'the filters defined in the melos.yaml for script $scriptName.';
+  }
+}
+
+class ScriptNotFoundException implements MelosException {
+  ScriptNotFoundException._(this.scriptName, this.availableScriptNames);
+
+  final String scriptName;
+  final List<String> availableScriptNames;
+
+  @override
+  String toString() {
+    final builder = StringBuffer(
+      'ScriptNotFoundException: The script $scriptName could not be found in the melos.yaml file.',
+    );
+
+    for (final scriptName in availableScriptNames) {
+      builder.write('\n - $scriptName');
+    }
+
+    return builder.toString();
+  }
+}
+
+class NoScriptException implements MelosException {
+  NoScriptException._();
+
+  @override
+  String toString() {
+    return "NoScriptException: This workspace has no scripts defined in it's 'melos.yaml' file.";
```

Comment:
Minor nit in that `'melos.yaml'` is single quoted here but not in the other exceptions above